scripts/apps/desks/directives/StageItemListDirective.js

```diff
@@ -206,12 +206,18 @@ export function StageItemListDirective(search, api, superdesk, desks, cards, $ti
                 }
             };
             scope.fetchPrevious = function() {
-                if (!scope.fetching && scope.page > 2) {
+                if (!scope.fetching && scope.page > 2 && criteria.source.from > 0) {
                     scope.fetching = true;
                     scope.page -= 1;
-                    criteria.source.from = scope.page > 3 ? (scope.page - 3) * criteria.source.size : 0;
                     scope.loading = true;
 
+                    if (scope.page > 3) {
+                        criteria.source.from = (scope.page - 2) * criteria.source.size;
+                    } else {
+                        criteria.source.size *= 2;
```

Comment:
This is only for `fetchPrevious`, for last two pages... it will go only once, when page is 3, it will fetch last 2 pages together... This is not set for `fetchNext`... For `fetchNext` it will go page by page.